wit/load.go

```diff
@@ -55,3 +55,33 @@ func LoadWIT(path string) (*Resolve, error) {
 
 	return DecodeJSON(&stdout)
 }
+
+// LoadWITFromBuffer loads WIT data from a provided buffer by processing it through wasm-tools.
+// It expects the buffer to contain valid WIT data and processes it through `wasm-tools`.
+// The result is returned as a *Resolve.
+func LoadWITFromBuffer(buffer *bytes.Buffer) (*Resolve, error) {
```

Comment:
Why does this accept a `bytes.Buffer` instead of `[]byte`?

Asking because I’ve generally seen `bytes.Buffer` used within a function, but not used as a return value or argument.

Reply:
yeah that's a good point, I will swtich it to `[]byte` instead